lib/_http_incoming.js

```diff
@@ -160,18 +160,27 @@ IncomingMessage.prototype._read = function _read(n) {
     readStart(this.socket);
 };
 
-
 // It's possible that the socket will be destroyed, and removed from
 // any messages, before ever calling this.  In that case, just skip
 // it, since something else is destroying this connection anyway.
-IncomingMessage.prototype.destroy = function destroy(error) {
-  // TODO(ronag): Implement in terms of _destroy
-  this.destroyed = true;
-  if (this.socket)
-    this.socket.destroy(error);
-  return this;
-};
+IncomingMessage.prototype._destroy = function _destroy(err, cb) {
+  if (!this.readableEnded || !this.complete) {
+    this.aborted = true;
+    this.emit('aborted');
+  }
 
+  // If aborted and the underlying socket not already destroyed,
+  // destroy it.
+  if (this.socket && !this.socket.destroyed && this.aborted) {
+    this.socket.destroy(err);
```

Comment:
@dnlup Question. What if `destroy(err || connResetException('aborted'))`? Then maybe we can remove `connResetException('aborted')` from the other 2 places.

Reply:
The problem I encountered with that approach is that some tests, like:

`test/parallel/test-async-wrap-tlssocket-asyncreset.js`

Are calling `destroy` without an error and don't expect an `error` event. They register a handler but check explicitly that it is not called. I interpreted this as "don't emit an error if none was passed".

Reply:
Ok. Let's leave it for now. Maybe a future semver-major.